lib/webdriveragent.js

```diff
@@ -96,12 +96,22 @@ class WebDriverAgent {
       }, this.log);
   }
 
+  /**
+   * Return true if the session does not need xcodebuild.
+   * @returns {boolean} Whether the session needs/has xcodebuild.
+   */
+  get canSkipXcodebuild () {
+    // Use this.args.webDriverAgentUrl to guarantee
+    // the capabilities set gave the `appium:webDriverAgentUrl`.
+    return this.usePreinstalledWDA || this.args.webDriverAgentUrl;
```

Comment:
if this.args defined?

Reply:
`this.args` is `{}` by default in this contractor, and no place to be null/undefined so not necessary?

Reply:
yes, a driver always has `this.args` set by basedriver

Reply:
> yes, a driver always has this.args set by basedriver

the WebDriverAgent class is not inherited from BaseDriver

Reply:
ah yes, good poitn